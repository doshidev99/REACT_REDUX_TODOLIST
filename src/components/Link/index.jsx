import React from 'react';
import PropTypes from 'prop-types';

const Link = ({ active, children, onClick }) => (
  <button type="button" onClick={onClick} disabled={active}>
    {children}
  </button>
);

Link.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.element,
  onClick: PropTypes.func
};

Link.defaultProps = {
  active: false,
  children: null,
  onClick: () => {}
};

export default Link;
