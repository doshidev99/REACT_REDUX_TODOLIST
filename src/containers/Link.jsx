import { connect } from 'react-redux';
import { setVisibilityFilterAction } from '../redux/filter/actions';
import Link from '../components/Link';

export default connect(
  (state, ownProps) => ({
    active: ownProps.filter === state.visibilityFilter
  }),
  (dispatch, ownProps) => ({
    onClick: () => {
      dispatch(setVisibilityFilterAction(ownProps.filter));
    }
  })
)(Link);
