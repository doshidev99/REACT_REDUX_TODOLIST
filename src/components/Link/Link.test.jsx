/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { shallow } from 'enzyme';
import Link from './index';

describe('Link', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Link debug />);

    expect(component).toMatchSnapshot();
  });
});
