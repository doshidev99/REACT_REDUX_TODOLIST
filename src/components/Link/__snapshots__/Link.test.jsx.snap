// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly in "debug" mode 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
